### ISSUE TITLE
Possibility for `--fusemount "host:..."`?

### ISSUE THREAD

**Issue by natefoo** [cvmfs/cvmfsexec#67]
I spent some time trying to get everything that `singcvmfs` does working on my own before I discovered this utility. It works great for containers that provide a reasonable runtime, but I'm having a few issues with some of the minimal [BioContainers](https://biocontainers.pro/) images. Many of these are based on busybox and don't provide a standard libc, etc. Even bind mounting in all the libs that libfuse3 needs (quite a few - The host's version of OpenSSL and a bunch of krb5 libs) doesn't work due to missing symbols in libc. I'm not sure, but it may possible to just bind mount all of `/usr/lib64` from the host system to some special path and then alter the dist libraries' rpath (or set `$LD_LIBRARY_PATH` in the wrapper).

There's a separate issue with `/var/run` and `/var/log` being a symlink to `/var/tmp` and thus breaking the scratch mount for `/var/run/cvmfs` and log mount for `/var/log/cvmfs`, but that should be more easily worked around if a solution can be found for the former.

Regardless, I am wondering whether `--fusemount "host:..."` rather than `--fusemount "container:..."` can be made to work, since it would solve the problem with the fuse3 and cvmfs libs running in the container runtime. I got pretty far with it before ultimately failing once discovering that a setuid `fusermount3` seemed to be required, but apparently not for container mode `--fusemount`?

**Comment by DrDaveD** [comment]
Yes that makes sense that a host fusermount3 is needed for `--fusemount "host:..."`.  It's not needed for `--fusemount "container:..."` because the kernel allows directly mounting fuse operating systems inside of a user namespace, without the assistance of any setuid-root helper.

**Comment by DrDaveD** [comment]
I think your second alternative is the probably the best.  Regular unprivileged apptainer/singularity works from cvmfs inside of cvmfsexec (and it is installed at /cvmfs/oasis.opensciencegrid.org/mis/apptainer/bin/apptainer).  cvmfsexec mode 3 does everything necessary to mount cvmfs within a root-mapped unprivileged user namespace at the correct `/cvmfs` mountpoint, and then starts another unprivileged user namespace mapping back to the original user.

singcvmfs was designed for installations that have only privileged singularity and no unprivileged user namespaces, and it turned out that it wasn't as generally useful as the cvmfsexec command.

**Comment by DrDaveD** [comment]
Yes it needs non-setuid singularity.  If you have a setuid installation, you can try adding a --userns option to test it.

**Comment by natefoo** [comment]
No dice - I guess (as per the documentation) `--userns` requires sandbox images so it'd be expected to fail when converting a docker image? That said, the oasis Apptainer version works great, so we can just use that:
```console
c506-061[skx](506)$ ~/bin/cvmfsexec oasis.opensciencegrid.org data.galaxyproject.org -- /cvmfs/oasis.opensciencegrid.org/mis/apptainer/bin/apptainer -s exec --cleanenv -B /cvmfs/data.galaxyproject.org:/cvmfs/data.galaxyproject.org:ro --home $HOME:$HOME docker://quay.io/biocontainers/blast:2.10.1--pl526he19e7b1_3 ls -l
 /cvmfs/data.galaxyproject.org
CernVM-FS: loading Fuse module... done
CernVM-FS: mounted cvmfs on /home1/03166/xcgalaxy/bin/.cvmfsexec/dist/cvmfs/oasis.opensciencegrid.org
CernVM-FS: loading Fuse module... done
CernVM-FS: mounted cvmfs on /home1/03166/xcgalaxy/bin/.cvmfsexec/dist/cvmfs/data.galaxyproject.org
total 9
drwxr-xr-x  210 xcgalaxy G-815688      4096 Apr 21  2022 byhand
drwxr-xr-x   18 xcgalaxy G-815688      4096 Nov 24  2020 managed
```

**Comment by DrDaveD** [comment]
Ok, since you have a plan I will close this issue.
